environs/config/config.go

```diff
@@ -1354,11 +1354,18 @@ func (c *Config) validateCharmHubURL() error {
 // Only three modes exist at the moment (strict or ""). Empty string
 // implies compatible mode.
 func (c *Config) Mode() ([]string, bool) {
-	if modes, ok := c.defined[ModeKey]; ok {
-		if m, ok := modes.([]string); ok {
-			return set.NewStrings(m...).SortedValues(), ok
+	modes, ok := c.defined[ModeKey]
+	if !ok {
+		return []string{}, false
+	}
+	if m, ok := modes.([]interface{}); ok {
+		s := set.NewStrings()
+		for _, v := range m {
+			s.Add(v.(string))
```

Comment:
not checking to see if v.(string) is successful: we know for sure this is a string?  not good if something else

Reply:
So we can be sure because it's validated first. I'm more than happy to add a check for this.